src/overlays/actors/ovl_Bg_Hidan_Rsekizou/z_bg_hidan_rsekizou.c

```diff
@@ -21,13 +20,193 @@ const ActorInit Bg_Hidan_Rsekizou_InitVars = {
+                                           yawSine * sphere->dim.modelSphere.center.z;
+        sphere->dim.worldSphere.center.y = (s16)this->dyna.actor.initPosRot.pos.y + sphere->dim.modelSphere.center.y;
+        sphere->dim.worldSphere.center.z =
+            (this->dyna.actor.initPosRot.pos.z - yawSine * sphere->dim.modelSphere.center.x) +
+            yawCosine * sphere->dim.modelSphere.center.z;
+    }
+
+    CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_8002F974(&this->dyna.actor, NA_SE_EV_FIRE_PILLAR - SFX_FLAG);
+}
+
+Gfx* BgHidanRsekizou_DrawFireball(GlobalContext* globalCtx, BgHidanRsekizou* this, s16 frame, MtxF* mf, s32 a,
+                                  Gfx* displayList) {
+    f32 coss;
+    f32 sins;
+    s32 temp;
+    f32 fVar6;
+    f32 tmpf7;
+
+    temp = (((this->burnFrame + frame) % 8) * 7) * 0.14285715f;
+    gSPSegment(displayList++, 9, SEGMENTED_TO_VIRTUAL(D_8088CD74[temp]));
+
+    fVar6 = (++frame != 4) ? frame + ((3 - this->blastFrame) * 0.33333334f) : frame;
+
+    gDPSetPrimColor(displayList++, 0, 0x01, 255, 255, 0, 150);
+    gDPSetEnvColor(displayList++, 255, 0, 0, 255);
+
+    if (a == 0) {
+        sins = -Math_Sins(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+        coss = -Math_Coss(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+    } else {
+        sins = Math_Sins(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+        coss = Math_Coss(this->dyna.actor.shape.rot.y - (frame * 0x5DC));
+    }
+
+    mf->xx = mf->yy = mf->zz = (0.7f * fVar6) + 0.5f; /* [0][0], [1][1], [2][2] */
+    tmpf7 = (((((0.7f * fVar6) + 0.5f) * 10.0f) * fVar6) + 20.0f);
+
+    mf->wx = (tmpf7 * sins) + this->dyna.actor.posRot.pos.x;                      /* [3][0] */
+    mf->wy = (this->dyna.actor.posRot.pos.y + 30.0f) + (0.699999988079f * fVar6); /* [3][1] */
+    mf->wz = (tmpf7 * coss) + this->dyna.actor.posRot.pos.z;                      /* [3][2] */
+
+    gSPMatrix(displayList++,
+              Matrix_MtxFToMtx(Matrix_CheckFloats(mf, "../z_bg_hidan_rsekizou.c", 543),
+                               Graph_Alloc(globalCtx->state.gfxCtx, 0x40)),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(displayList++, D_0600DC30);
+
+    return displayList;
+}
+
+void BgHidanRsekizou_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanRsekizou* this = THIS;
+    s32 i;
+    s32 pad;
+    MtxF mf;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
+
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_hidan_rsekizou.c", 564);
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_hidan_rsekizou.c", 568),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(gfxCtx->polyOpa.p++, &D_0600AD00);
+    Matrix_MtxFCopy(&mf, &gMtxFClear); /* Set an Identity Matrix */
```

Comment:
The matrix being copied is already called clear to describe the purpose of this kind of operation, so the comment is unnecessary imo.
```suggestion
    Matrix_MtxFCopy(&mf, &gMtxFClear);
```